### DISCUSSION TITLE
Proposal: [DebuggerBrowsable(DebuggerBrowsableState.Never)] on Record's EqualityContract

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4001] [Language Ideas]
I would like to suggest that ```DebuggerBrowsable(DebuggerBrowsableState.Never)]``` be applied to Record's ```EqualityContract``` member so that EqualityContract does not clutter the view when Watching/Inspecting record classes.

**Comment by CyrusNajmabadi** [comment]
@jcouv for visibility.  I don't really love this idea.  But i'm not strongly opposed to it either.  I just don't really like the idea of emitting code by default just for a VS IDE scenario (esp. around a feature that we may change at any point in time).

**Comment by YairHalberstadt** [comment]
Wouldn't it be better for a record to emit some kind of marker so VS can tell it's a record, and choose its own policy to apply?

See https://github.com/dotnet/roslyn/issues/45777

**Comment by CyrusNajmabadi** [comment]
> Wouldn't it be better for a record to emit some kind of marker so VS can tell it's a record, and choose its own policy to apply?

IMO, yes.  And we *do* have a PR where this was done.  i.e. the Roslyn API will expose `.IsRecord` to know if something is a record.  So we could decide if we should hide some details of it.

**Comment by tmat** [comment]
@CyrusNajmabadi Debugger* attributes are commonly emitted by the compiler in many scenarios. It's not for VS IDE - it's for debugging experience in general (not just in VS). 
The compiler should absolutely consider debugging experience when implementing language features.

**Comment by CyrusNajmabadi** [comment]
Sorry, i read this as EditorBrowsable.

**Comment by tmat** [comment]
@CyrusNajmabadi Yeah, I'd agree with that.
